packages/connection/legacy/class-jetpack-xmlrpc-server.php

```diff
@@ -751,8 +753,8 @@ public function unlink_user( $user_id = null ) {
 		 */
 		do_action( 'jetpack_event_log', 'unlink' );
 		return Connection_Manager::disconnect_user(
-			empty( $user_id ) ? null : $user_id,
-			empty( $user_id ) ? false : true
+			$user_id,
+			! (bool) $user_id
```

Comment:
Earlier this was `empty( $user_id ) ? false : true`, an empty array or 0 would pass false.

Now you have `! (bool) $user_id`, an empty array or 0 will pass true. Given the docs above, I think this is now backwards.

Reply:
oops. I was going to argue that the former syntax was easier to read and avoided errors like this. But this is not a justification ;) Fixed.